packages/app/src/vis-packs/nexus/models.ts

```diff
@@ -44,16 +43,16 @@ export interface NxValues<T extends NumericType | ComplexType> {
   signal: ArrayValue<T>;
   signalLabel: string;
   errors?: NumArray;
-  axisMapping: AxisMapping;
+  axisValues: AxisMapping<NumArray>;
   auxiliaries?: Auxiliary[];
   title: string;
 }
 
-export type AxisDatasetMapping = (NumArrayDataset | undefined)[];
+export type AxisMapping<T> = (T | undefined)[];
 
 export interface SilxStyle {
   signalScaleType?: ScaleType;
-  axisScaleTypes?: ScaleType[];
+  axisScaleTypes?: AxisMapping<ScaleType>;
```

Comment:
I think this property was incorrectly typed, by the way, as the array can contain `undefined` values, right? Hmm, looking at `getSilxStyle`, I think we actually have a bug there, as we disregard `axisScaleTypes` arrays that contain `undefined` values.

Reply:
Which would change the correspondence between position and axis of the scale types ?

To be checked with a dedicated dataset in mock.

Reply:
If there's `null` in the `axisScaleTypes` array, we act as if there is no array at all. So the bug is more that we ignore a `SILX_style` info when we could easily use it. I'll open an issue.